src/irr/asset/CGLSLBRDFBuiltinIncludeLoader.h

```diff
@@ -417,6 +438,24 @@ float irr_glsl_ggx_trowbridge_reitz(in float a2, in float NdotH2)
     return a2 / (irr_glsl_PI * denom*denom);
 }
 
+BSDFSample irr_glsl_ggx_cos_gen_sample(in uvec2 _sample, in float _a2)
+{
+    vec2 u = vec2(_sample)/float(4294967295);
+    float phi = 2.0*irr_glsl_PI*u.x;
+    float cos2Theta = (1.0 - u.y)/(u.y*(_a2 - 1.0) + 1.0);
+    float cosTheta = sqrt(cos2Theta);
+    float sinTheta = sqrt(1.0 - cos2Theta);
+    
+    BSDFSample smpl;
+    //erm... actually i'm returning half vector here, idk how could i get L with this api
+    //user should read this H (half vector/micro-surface normal) and compute L by reflecting V on H
+    //just remember that returned vector is in reflection space so first it has to be transformed to space in which light computations are done
+    smpl.L = vec3(sinTheta,sinTheta,cosTheta)*vec3(cos(phi),sin(phi),1.0);
+    smpl.probability = irr_glsl_ggx_trowbridge_reitz(_a2, cos2Theta)*abs(cosTheta);//multiply by cosTheta in order for the NDF to actually be PDF (integrates to 1)
+
+    return smpl;
+}
```

Comment:
no use for just sampling the NDF, just give me an isotropic roughness importance sampling (with fresnel)

Reply:
anisotropic as well